src/features/account/controllers/account-controller.test.ts

```diff
@@ -7,47 +7,41 @@ import { accountDeleteBySchema } from '@/features/account/validators/account-fin
 import { HttpError } from '@/shared/errors/http-error';
 import { HttpStatusCode } from '@/shared/protocols/http-client';
 
-const makeSut = () => {
-  const deleteUserAccountServiceMock = mock<DeleteUserAccountsService>({
-    execute: vi.fn(),
-  });
-
-  const accountController = new AccountController(deleteUserAccountServiceMock);
-
-  const req = mockDeep<Request>();
-
-  const res = {
-    send: vi.fn(),
-
-    status: vi.fn().mockReturnThis(),
-  } as unknown as Response;
+describe('deleteAccountById', () => {
+  let deleteUserAccountServiceMock: DeleteUserAccountsService;
 
-  const next = vi.fn();
+  let accountController: AccountController;
 
-  return {
-    accountController,
+  let req: Request;
+  let res: Response;
+  let next: NextFunction;
+  let error: HttpError;
 
-    deleteUserAccountService: deleteUserAccountServiceMock,
+  beforeEach(() => {
+    deleteUserAccountServiceMock = mock<DeleteUserAccountsService>({
+      execute: vi.fn(),
+    });
+    accountController = new AccountController(deleteUserAccountServiceMock);
 
-    next,
+    req = mockDeep<Request>();
 
-    req,
+    res = {
+      json: vi.fn(),
+      send: vi.fn(),
+      status: vi.fn().mockReturnThis(),
+    } as unknown as Response;
```

Comment:
Tava dando uma estudada no seu codigo pra aprender sobre testes e vi que da pra fazer uma melhoria aqui.
```suggestion
res = mockDeep<Response>({
  json: vi.fn(),
  send: vi.fn(),
  status: vi.fn().mockReturnThis(),
});
```

ve se faz sentido pra você, ainda to estudando. tmj bb

Reply:
Isso é pra ele conseguir usar o res como mock, não é a melhor maneira, mas é a que funciona